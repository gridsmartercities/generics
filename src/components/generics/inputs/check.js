import React from "react";
import PropTypes from "prop-types";

const CheckInput = ({ label, name, type = "checkbox", checked }) => (
  <div>
    <label htmlFor={name}>{label}</label>
    <input type={type} id={name} name={name} checked={checked} />
  </div>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default CheckInput;
