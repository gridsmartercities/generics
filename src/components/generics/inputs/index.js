import InputComponent from "./input";
import CheckInputComponent from "./input";

export const Input = InputComponent;
export const CheckInput = CheckInputComponent;
