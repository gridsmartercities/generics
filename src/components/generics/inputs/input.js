import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Label from "../label";
const InputStyle = styled.input`
  border: ${({ theme }) => theme.input.border || "1px solid #ccc"};
  padding: ${({ theme }) => theme.input.padding || "0.5rem"};
  margin: ${({ theme }) => theme.input.margin || "5px 0"};
  background: ${({ theme }) => theme.input.background || "#fff"};
  border-radius: ${({ theme }) => theme.input.borderRadius || "5px"};
`;

const Input = props => {
  const { label, name } = props;
  return (
    <div>
      <Label htmlFor={name}>{label}</Label>
      <InputStyle {...props} id={name} />
    </div>
  );
};

Input.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Input;
