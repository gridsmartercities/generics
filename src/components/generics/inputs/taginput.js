import React, { useState, useRef } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Label from "../label";

const InputStyle = styled.div`
  display: block;
  width: 100%;
  box-sizing: border-box;
  border: ${({ theme }) => theme.input.border || "1px solid #ccc"};
  padding: ${({ theme }) => theme.input.padding || "0.5rem"};
  margin: ${({ theme }) => theme.input.margin || "5px 0"};
  background: ${({ theme }) => theme.input.background || "#fff"};
  border-radius: ${({ theme }) => theme.input.borderRadius || "5px"};
`;

const Tag = styled.div`
  background: red;
`;

const Input = props => {
  const [tags, updateTags] = useState([]);
  const input = useRef();
  const update = e => {
    if (e.keyCode === 13) {
      updateTags(e.target.innerText.split(","));
    }
  };
  const remove = e => {
    const text = e.target.dataset.entry;
    console.log(text);
    const newTags = tags.filter(item => item !== text);
    input.current.innerText = newTags.join(",");
    updateTags(newTags);
  };
  return (
    <>
      <input type="hidden" value={JSON.stringify(tags)} />
      {tags.map(item => (
        <Tag>
          {item}
          <span data-entry={item} onClick={remove}>
            X
          </span>
        </Tag>
      ))}
      <InputStyle ref={input} contentEditable={true} onKeyDown={update} />
    </>
  );
};

export default Input;
