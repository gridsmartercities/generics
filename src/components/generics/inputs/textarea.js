import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Label from "../label";

const TextareaInput = styled.textarea`
  display: block;
  width: 100%;
  box-sizing: border-box;
  border: ${({ theme }) => theme.input.border || "1px solid #ccc"};
  padding: ${({ theme }) => theme.input.padding || "0.5rem"};
  margin: ${({ theme }) => theme.input.margin || "5px 0"};
  background: ${({ theme }) => theme.input.background || "#fff"};
  border-radius: ${({ theme }) => theme.input.borderRadius || "5px"};
`;

const Textarea = props => {
  const { label, name } = props;
  return (
    <div>
      <Label htmlFor={name}>{label}</Label>
      <TextareaInput {...props} id={name} />
    </div>
  );
};

Textarea.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Textarea;
