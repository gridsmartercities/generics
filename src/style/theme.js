const theme = {
  main: {},
  input: {},
  label: {},
  buttons: {},
  containers: {}
};

export default theme;
