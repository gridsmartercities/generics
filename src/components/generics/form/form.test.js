import "jest-dom/extend-expect";
import React from "react";
import { Simulate } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import Form from "./index";
import { Input } from "../inputs";
import theme from "../../../style/theme";
import { ThemeProvider } from "styled-components";

test("Form Component", () => {
  const submitAction = jest.fn(data => data);
  const { getByLabelText, container } = render(
    <ThemeProvider theme={theme}>
      <Form onSubmit={submitAction} buttonValue="Send">
        <Input label="test 1" name="test1" />
        <Input label="test 2" name="test2" />
      </Form>
    </ThemeProvider>
  );

  const formNode = document.querySelector("form");
  const input1 = getByLabelText("test 1");
  const input2 = getByLabelText("test 2");
  input1.value = "gary";
  input2.value = "blue";

  const button = formNode.querySelector("button");

  Simulate.submit(formNode);
  expect(submitAction.mock.results[0].value).toEqual({
    test1: "gary",
    test2: "blue"
  });
  expect(button).toHaveTextContent("Send");
  expect(submitAction).toHaveBeenCalledTimes(1);
  expect(container.firstChild).toMatchSnapshot();
});
