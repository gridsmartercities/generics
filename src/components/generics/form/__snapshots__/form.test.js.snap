// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component 1`] = `
<form>
  <label
    for="test1"
  >
    test 1
  </label>
  <input
    id="test1"
    name="test1"
  />
  <label
    for="test2"
  >
    test 2
  </label>
  <input
    id="test2"
    name="test2"
  />
  <button
    class="sc-bwzfXH jjWYyR"
    type="submit"
  >
    Send
  </button>
</form>
`;
